Sofa/Component/Constraint/Lagrangian/Model/src/sofa/component/constraint/lagrangian/model/FixedLagrangianConstraint.h

```diff
@@ -0,0 +1,96 @@
+*                                                                             *
+* This program is free software; you can redistribute it and/or modify it     *
+* under the terms of the GNU Lesser General Public License as published by    *
+* the Free Software Foundation; either version 2.1 of the License, or (at     *
+* your option) any later version.                                             *
+*                                                                             *
+* This program is distributed in the hope that it will be useful, but WITHOUT *
+* ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or       *
+* FITNESS FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License *
+* for more details.                                                           *
+*                                                                             *
+* You should have received a copy of the GNU Lesser General Public License    *
+* along with this program. If not, see <http://www.gnu.org/licenses/>.        *
+*******************************************************************************
+* Authors: The SOFA Team and external contributors (see Authors.txt)          *
+*                                                                             *
+* Contact information: contact@sofa-framework.org                             *
+******************************************************************************/
+#pragma once
+#include <sofa/component/constraint/lagrangian/model/config.h>
+
+#include <sofa/core/behavior/Constraint.h>
+#include <sofa/core/behavior/ConstraintResolution.h>
+#include <sofa/core/behavior/MechanicalState.h>
+#include <sofa/core/behavior/OdeSolver.h>
+#include <sofa/core/topology/TopologySubsetIndices.h>
+
+namespace sofa::component::constraint::lagrangian::model
+{
+
+template< class DataTypes >
+class FixedLagrangianConstraint : public core::behavior::Constraint<DataTypes>
+{
+public:
+    SOFA_CLASS(SOFA_TEMPLATE(FixedLagrangianConstraint,DataTypes), SOFA_TEMPLATE(core::behavior::Constraint,DataTypes));
+
+    typedef typename DataTypes::VecCoord VecCoord;
+    typedef typename DataTypes::VecDeriv VecDeriv;
+    typedef typename DataTypes::Coord Coord;
+    typedef typename DataTypes::Deriv Deriv;
+    typedef typename DataTypes::MatrixDeriv MatrixDeriv;
+    typedef typename Coord::value_type Real;
+    typedef typename core::behavior::MechanicalState<DataTypes> MechanicalState;
+    typedef typename core::behavior::Constraint<DataTypes> Inherit;
+
+    typedef typename DataTypes::MatrixDeriv::RowIterator MatrixDerivRowIterator;
+    typedef core::objectmodel::Data<VecCoord>		DataVecCoord;
+    typedef core::objectmodel::Data<VecDeriv>		DataVecDeriv;
+    typedef core::objectmodel::Data<MatrixDeriv>    DataMatrixDeriv;
+
+    typedef sofa::core::topology::TopologySubsetIndices SetIndex;
+
+
+protected:
+
+    std::vector<unsigned int> m_cid;
+    std::vector<Deriv> m_prevForces;
```

Comment:
```suggestion
    sofa::type::vector<unsigned int> m_cid;
    sofa::type::vector<Deriv> m_prevForces;
```

Reply:
According to this : https://github.com/sofa-framework/sofa/issues/4217 there was a need to use less custom container and rely more on the STL. Is there a real interest in using the sofa vector ?

Reply:
Today, there the interesting feature is the out-of-bounds check in Debug mode. But I agree with you that one day, `sofa::type::vector` must be removed